### PR TITLE
Add tests for RemoteBmi.jl

### PR DESCRIPTION
**Pull request by sverhoeven**
Run tests with
```shell
julia -e 'using Pkg; Pkg.activate("."); Pkg.test()'
```

TODO
* [x] get a test working
* [x] Move client from test to src
* [x] Make client have https://github.com/Deltares/BasicModelInterface.jl interface
* [x] Use openapi generated julia client in tests, started in #16
* [x] test all bmi methods
* [ ] test invalid requests
* [x] run tests in ci (use [BestieTemplate](https://github.com/JuliaBesties/BestieTemplate.jl))
* [x] document tests

### CODE REVIEW

**Comment by sverhoeven** [comment]
Using [act](https://nektosact.com/introduction.html) was able to run `act -W  ./.github/workflows/julia-test-on-prs.yml` until codecov step OK.

**Comment by BSchilperoort** [comment]
Tests run fine, but I can't get the julia server to work with the heat model:

```sh
(@v1.9) pkg> dev /home/bart/git/remotebmi/RemoteBMI.jl
   Resolving package versions...
    Updating `~/.julia/environments/v1.9/Project.toml`
  [4b6948d5] + RemoteBMI v0.1.0 `~/git/remotebmi/RemoteBMI.jl`
    Updating `~/.julia/environments/v1.9/Manifest.toml`
  [d1d4a3ce] + BitFlags v0.1.9
  [944b1d66] + CodecZlib v0.7.6
  [f0e56b4a] + ConcurrentUtilities v2.4.2
  [460bff9d] + ExceptionUnwrapping v0.1.10
  [e2ba6199] + ExprTools v0.1.10
  [cd3eb016] + HTTP v1.10.9
  [842dd82b] + InlineStrings v1.4.2
  [682c06a0] + JSON v0.21.4
  [6c6e2e6c] + MIMEs v0.1.4
  [739be429] + MbedTLS v1.1.9
  [78c3b35d] + Mocking v0.8.1
  [d5e62ea6] + OpenAPI v0.1.26
  [4d8831e6] + OpenSSL v1.4.3
  [69de0a69] + Parsers v2.8.1
  [4b6948d5] + RemoteBMI v0.1.0 `~/git/remotebmi/RemoteBMI.jl`
  [6c6a2e73] + Scratch v1.2.1
  [777ac1f9] + SimpleBufferStream v1.2.0
  [dc5dba14] + TZJData v1.3.0+2024b
  [f269a46b] + TimeZones v1.19.0
  [3bb67fe8] + TranscodingStreams v0.11.3
  [5c2747f8] + URIs v1.5.1

julia> import RemoteBMI.Server: run
[ Info: Precompiling RemoteBMI [4b6948d5-498c-54b5-b082-cdb1540453e9]

julia> using Heat
[ Info: Precompiling TimeZonesRecipesBaseExt [7f969005-de8d-58b6-a711-4f863f324b87]

julia> port = parse(Int, get(ENV, "BMI_PORT", "50051"))
50051

julia> run(Heat.Model, "0.0.0.0", port)
ERROR: MethodError: no method matching run(::Type{Heat.Model}, ::String, ::Int64)

Closest candidates are:
  run(::Base.AbstractCmd, ::Any...; wait)
   @ Base process.jl:477

Stacktrace:
 [1] top-level scope
   @ REPL[7]:1

```

**Comment by sverhoeven** [comment]
Oeps forgot to rename run to run_bmi_server in example. Can you try again?

**Comment by BSchilperoort** [comment]
> Oeps forgot to rename run to run_bmi_server in example. Can you try again?

I did. It mostly works, encountered some issues: #18 #19 #20

File: RemoteBMI.jl/test/fake.jl [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+Base.@kwdef mutable struct Model
+  values::Vector{Float64} = [1.0, 2.0, 3.0, 4.0]
+  current::Float64 = 0.0
+end
+
+BMI.initialize(::Type{Model}, config_file) = Model()
+function BMI.update(m::Model)
+  m.current += BMI.get_time_step(m)
+  return nothing
```
Comment (BSchilperoort) [inline RemoteBMI.jl/test/fake.jl:13]: Should return `m` here.

**Pull request merged**